crates/sui-core/src/consensus_handler.rs

```diff
@@ -887,23 +889,45 @@ impl TransactionManagerSender {
 
     fn send(
         &self,
-        transactions: Vec<VerifiedExecutableTransaction>,
+        transactions: Vec<Schedulable>,
+        assigned_versions: AssignedTxAndVersions,
         scheduling_source: SchedulingSource,
     ) {
-        let _ = self.sender.send((transactions, scheduling_source));
+        let _ = self
+            .sender
+            .send((transactions, assigned_versions, scheduling_source));
     }
 
     async fn run(
         mut recv: monitored_mpsc::UnboundedReceiver<(
-            Vec<VerifiedExecutableTransaction>,
+            Vec<Schedulable>,
+            AssignedTxAndVersions,
             SchedulingSource,
         )>,
         execution_scheduler: Arc<ExecutionSchedulerWrapper>,
         epoch_store: Arc<AuthorityPerEpochStore>,
     ) {
-        while let Some((transactions, scheduling_source)) = recv.recv().await {
+        while let Some((transactions, assigned_versions, scheduling_source)) = recv.recv().await {
             let _guard = monitored_scope("ConsensusHandler::enqueue");
-            execution_scheduler.enqueue(transactions, &epoch_store, scheduling_source);
+            let assigned_versions = assigned_versions.into_map();
+            let txns = transactions
+                .into_iter()
+                .map(|txn| {
+                    let key = txn.key();
+                    (
+                        txn,
+                        ExecutionEnv {
```

Comment:
why not use the builder pattern you implemented? a bit confusing to see a mix of both styles